te_dp/Makefile

```diff
@@ -39,20 +39,20 @@ BINDIR = bin
 DP_SRC   = $(wildcard $(SRCDIR)/*.c)
 DP_OBJ   = $(DP_SRC:$(SRCDIR)/%.c=$(OBJDIR)/%.o)
 DP_BIN   = $(BINDIR)/te_dp
-DP_FLAGS = -lcurl -luv -ljson -lssl -lpthread
+DP_FLAGS = -lcurl -luv -ljson-c -lssl -lpthread
 
 STAT_SRC   = $(wildcard $(SRCDIR)/*.cpp)
 STAT_OBJ   = $(STAT_SRC:$(SRCDIR)/%.cpp=$(OBJDIR)/%.o)
 STAT_FLAGS = -lzmq -ljson
 STAT_BIN   = $(BINDIR)/te_stats_collector
 
 CFLAGS     = -g -O0 -Wall -I.
-STAT_FLAGS = -pthread -lzmq -ljson
+STAT_FLAGS = -pthread -lzmq -ljson-c
 
 all: dp stat
 
 dp: $(DP_OBJ)
-	@$(CC) $(CFLAGS) -o $(DP_BIN) $^ $(DP_FLAGS)
+	@$(CC) $(CFLAGS) -o $(DP_BIN) $^ $(DP_FLAGS) -no-pie
```

Comment:
Issue seen :- without `-no-pie`  
https://stackoverflow.com/questions/73275183/gcc-assembly-x86-relocation-warning-creating-dt-textrel-in-a-pie

Further reference:- 
https://stackoverflow.com/a/50615370 -> read comments as well